app/controllers/groups/samples_controller.rb

```diff
@@ -63,21 +62,32 @@ def current_page
       @current_page = t(:'groups.sidebar.samples')
     end
 
-    def set_default_sort
-      # remove metadata sort if metadata not visible
-      if !@q.sorts.empty? && @q.sorts[0].name.start_with?('metadata_') && params[:q][:metadata].to_i != 1
-        @q.sorts.slice!(0)
-      end
+    def set_metadata_fields
+      fields_for_namespace(namespace: @group, show_fields: @search_params && @search_params[:metadata].to_i == 1)
+    end
+
+    def process_samples
+      authorize! @group, to: :sample_listing?
+      @search_params = search_params
+      set_metadata_fields
 
-      @q.sorts = 'updated_at desc' if @q.sorts.empty?
+      @q = authorized_samples.ransack(@search_params)
+      @pagy, @samples = pagy_with_metadata_sort(@q.result)
```

Comment:
**nitpick (blocking):** Can you move the last 2 lines of this method into the index as it is not required for `search`

Reply:
Moved in 8ffdb7678b9b118ee9c4705612df765ea501a048